### PR TITLE
[MetricBeat] resource tags map should be compatible with short or whole resource id

### LINKED ISSUE #20326: [MetricBeat] [AWS]   events can't set TargetGroup tags
For confirmed bugs, please report:
- Version: 7.8.1
- Steps to Reproduce: 

  * set aws ALB/NLB targetgroup tags
  * run metricbeat with this example config 

```yaml
metricbeat.modules:
  - module: aws
    aws_partition: aws
    period: 300s
    access_key_id:  'your aws key'
    secret_access_key: 'your aws key'
    metricsets:
      - cloudwatch
    metrics:
      - namespace: AWS/ApplicationELB
        statistic: ['Minimum']
        name: ['HealthyHostCount']
        tags.resource_type_filter: elasticloadbalancing
```

*  current result:

events only contain Loadbalancer tags

* expected result:

events contain both LoadBalancer and TargetGroup tags 

----

aws alb and nlb metrics structure

```json
// CLI output ： aws resourcegroupstaggingapi get-resources --resource-type-filters elasticloadbalancing
        {
            "Namespace": "AWS/ApplicationELB",
            "MetricName": "UnHealthyHostCount",
            "Dimensions": [
                {
                    "Name": "TargetGroup",
                    "Value": "targetgroup/tg-name/12345678abc"
                },
                {
                    "Name": "LoadBalancer",
                    "Value": "app/lb-name/12345678abc"
                }
            ]
        },

        {
            "Namespace": "AWS/NetworkELB",
            "MetricName": "HealthyHostCount",
            "Dimensions": [
                {
                    "Name": "TargetGroup",
                    "Value": "targetgroup/tg-name/012345678abcdef"
                },
                {
                    "Name": "LoadBalancer",
                    "Value": "net/lb-name/012345678abcdef"
                }
            ]
        },
```

* aws resourcetagginggroup result examle

```


{
    "ResourceARN": "arn:aws:elasticloadbalancing:ap-northeast-2:123456789012:loadbalancer/app/alb-name/12345678abc",
        "Tags": [
        {
            "Key": "K1",
            "Value": "V1"
        },
}
{
    "ResourceARN": "arn:aws:elasticloadbalancing:ap-northeast-2:123456789012:loadbalancer/net/nlb-name/2233445566",
        "Tags": [
        {
            "Key": "K2",
            "Value": "V2"
        },
}

{
    "ResourceARN": "arn:aws:elasticloadbalancing:ap-northeast-2:123456789012:targetgroup/tg-name/012345678abcdef"
        "Tags": [
        {
            "Key": "K3",
            "Value": "V3"
        },
}

```

comparing above two example results, we can figure out that ALB and NLB `LoadBalancer` dimension value is not same with `resource id` field of ARN, and `TargetGroup` dimension value is same with `resource id` field of ARN.

* the reason

`GetResourcesTags` func has proccessed the resourcegroupstaggingapi result, the `resourceTagMap` is like this, so we can not get TargetGroup tags in `insertTags` function .

```
{
   "tg-name/012345678abcdef":   {
            "Key": "K3",
            "Value": "V3"
        },
  "net/nlb-name/2233445566" :  {
            "Key": "K2",
            "Value": "V2"
   }
}
```

https://github.com/elastic/beats/blob/ea63f053e54eaf50ed02b736686ca41f3289bc00/x-pack/metricbeat/module/aws/utils.go#L189-L197

* suggestion

 I think we should use both short resource identifier and whole resource identifier of ARN, just like this

```
{
   "tg-name/012345678abcdef":   {
            "Key": "K3",
            "Value": "V3"
        },
  "net/nlb-name/2233445566" :  {
            "Key": "K2",
            "Value": "V2"
   },

   "targetgroup/tg-name/012345678abcdef":   {
            "Key": "K3",
            "Value": "V3"
        },
  "loadbalancer/net/nlb-name/2233445566" :  {
            "Key": "K2",
            "Value": "V2"
   }
}
```

### PR DESCRIPTION
**Pull request by kwinstonix**
- Bug

## What does this PR do?

fix resource tags in metircbeat aws cloudwatch module.  more detail has be explained in #20326

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

1.  metricbeat aws module config

```yaml
metricbeat.modules:
  - module: aws
    aws_partition: aws
    period: 300s
    access_key_id: 'your key'
    secret_access_key: 'your key'
    regions:
      - ap-northeast-2
    metricsets:
      - cloudwatch
    metrics:
      - namespace: AWS/ApplicationELB
        statistic: ['Minimum']
        name: ['HealthyHostCount']
        #tags.resource_type_filter: elasticloadbalancing   # for verion 7.x
        resource_type: elasticloadbalancing     # for version 8.0

```

2. set AWS ELB TargetGroup tags

3. run metricbeat,  `aws.tags` field in event should contain both LoadBalancer and TargetGroup tags

## Related issues

- Closes #20326

### CODE REVIEW

**Comment by kwinstonix** [comment]
@kaiyan-sheng , could you help me review this ?  😊

File: x-pack/metricbeat/module/aws/utils_test.go [hunk 1]
```diff
@@ -377,7 +377,7 @@ func TestFindIdentifierFromARN(t *testing.T) {
 	}
 
 	for _, c := range cases {
-		identifier, err := FindIdentifierFromARN(c.resourceARN)
+		identifier, err := FindShortIdentifierFromARN(c.resourceARN)
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/utils_test.go:380]: Can we also add tests for `FindWholeIdentifierFromARN` here please?
```suggestion
		shortIdentifier, err := FindShortIdentifierFromARN(c.resourceARN)
		assert.NoError(t, err)
		assert.Equal(t, c.expectedShortIdentifier, shortIdentifier)

		wholeIdentifier, err := FindWholeIdentifierFromARN(c.resourceARN)
		assert.NoError(t, err)
		assert.Equal(t, c.expectedWholeIdentifier, wholeIdentifier)
```

**Comment by kaiyan-sheng** [comment]
Hi @kwinstonix ! SORRY for the delay on reviewing this PR 😂 Overall it looks good! I only added one small comment. Also could you rebase this PR please? Thank you!

File: x-pack/metricbeat/module/aws/utils_test.go (hunk 1 above)
Comment (kwinstonix) [reply x-pack/metricbeat/module/aws/utils_test.go]: yes, It is done

**Comment by kwinstonix** [comment]
The PR is rebased on master

File: x-pack/metricbeat/module/aws/utils_test.go (hunk 1 above)
Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/aws/utils_test.go]: Thank you!

**Comment by kaiyan-sheng** [comment]
jenkins run the tests

**Pull request merged**